### PR TITLE
cmake: fix openssl detection

### PR DESCRIPTION
**Pull request by furiel**
- FindOpenSSL.cmake provides  OPENSSL_INCLUDE_DIR instead of OPENSSL_INCLUDE_DIRS
(less /usr/share/cmake-3.5/Modules/FindOpenSSL.cmake)

- Adding missing detection for openssl functions that were provided by autotools.

- Hinting cmake:
Checkout and install openssl to $pwd_openssl_root.
cmake -DOPENSSL_ROOT_DIR=$pwd_openssl_root
Tested with openssl 1.1.0g

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

File: lib/CMakeLists.txt [hunk 1]
```diff
@@ -286,6 +286,23 @@ add_library(syslog-ng SHARED ${LIB_SOURCES})
...
 target_include_directories(syslog-ng SYSTEM INTERFACE ${CORE_INCLUDE_DIRS})
 
+function (openssl_set_define_for_symbol target symbol)
+  set (CMAKE_REQUIRED_INCLUDES ${OPENSSL_INCLUDE_DIR})
+  set (CMAKE_REQUIRED_LIBRARIES ${OPENSSL_LIBRARIES})
+  include(CheckSymbolExists)
+  check_symbol_exists(${symbol} "openssl/evp.h;openssl/ssl.h;openssl/x509v3.h" HAVE_${symbol})
+  if (HAVE_${symbol})
+    string(TOUPPER SYSLOG_NG_HAVE_DECL_${symbol} SYMBOL_WITH_CAPITALS)
+    target_compile_definitions(syslog-ng PUBLIC ${SYMBOL_WITH_CAPITALS}=1)
```
Comment (Kokan) [inline lib/CMakeLists.txt]: The `SYSLOG_NG_HAVE...` compiler switch variables are provided via a header file `syslog-ng-config.in.h`. There is no need to directly set it with `target_compile_definitions`.

File: lib/CMakeLists.txt [hunk 2]
```diff
@@ -286,6 +286,23 @@ add_library(syslog-ng SHARED ${LIB_SOURCES})
...
+  endif()
+endfunction()
+
+openssl_set_define_for_symbol(syslog-ng EVP_MD_CTX_reset)
```
Comment (Kokan) [inline lib/CMakeLists.txt]: This could be moved to the section where the rest of the `check_symbol_exists` are.

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Pull request merged**